colossus/src/main/scala/colossus/core/Worker.scala

```diff
@@ -238,25 +238,25 @@ private[colossus] class Worker(config: WorkerConfig) extends Actor with ActorLog
         connection.serverHandler.shutdownRequest()
       }}
     }
-    case DelegatorMessage(server, message) => {
-      delegators
-        .find{case (_, delegator) => delegator.server == server}
-        .map{case (_, delegator) =>
-          delegator.handleMessage.orElse[Any, Unit] {
-            case unhandled => log.warning(s"Unhandled message $unhandled for delegator of server ${server.name}")
+    case InitializerMessage(server, message) => {
+      initializers
+        .find{case (_, initializer) => initializer.server == server}
+        .map{case (_, initializer) =>
+          initializer.receive.orElse[Any, Unit] {
+            case unhandled => log.warning(s"Unhandled message $unhandled for initializer of server ${server.name}")
           }(message)
         }
         .getOrElse{
-          log.error(s"delegator message $message for unknown server ${server.name}")
+          log.error(s"initializer message $message for unknown server ${server.name}")
         }
     }
-    case NewConnection(sc, attempt) => delegators.get(sender()).map{delegator =>
-      delegator.createConnectionHandler.map{handler =>
-        registerConnection(sc, delegator.server, handler)
+    case NewConnection(sc, attempt) => initializers.get(sender()).map{initializer =>
+      Try(initializer.onConnect.apply(null)).map{handler =>
```

Comment:
1.  This obviously needs to be fixed (this is where the `ServerContext` should be generated.
2.  It's a little disturbing that all tests pass even with this code.